pinot-common/src/main/java/com/linkedin/pinot/common/request/transform/TransformFunctionFactory.java

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.common.request.transform;
+
+import com.google.common.base.Preconditions;
+import java.util.HashMap;
+import java.util.Map;
+import javax.annotation.Nonnull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Factory class for transformation functions.
+ */
+public class TransformFunctionFactory {
+  private static final Logger LOGGER = LoggerFactory.getLogger(TransformFunctionFactory.class);
+
+  private static final Map<String, TransformFunction> TRANSFORM_FUNCTION_MAP = new HashMap<>();
+  private static boolean _inited = false;
+
+  /**
+   * Private constructor, to prevent instantiation.
+   */
+  private TransformFunctionFactory() {
+
+  }
+
+  /**
+   * This method builds the transform factory containing functions
+   * specified in the server configuration. Throws {@link RuntimeException} if it has
+   * already been initialized. Method is synchronized (as opposed to concurrent), as it is
+   * expected to be called only once, during start-up.
+   *
+   * @param transformFunctions Array of transform function names
+   */
+  @SuppressWarnings("unchecked")
+  public static synchronized void init(@Nonnull String[] transformFunctions) {
```

Comment:
If init() is synchronized, then get() should be synchronized as well? Or at least put "_inited = true" as the last statement in this method?
Also, it looks a little weird to make the init() a static method. Since we need to pass in the transformFunctions, why not just hardcode a map for this class?